### PR TITLE
cmd: Fix --authfile/--insecure flags

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
d1bccf6c124f ("gadgets/run: Support eBPF parameters") introduced a logic to remove the flags before calling GetGadgetInfo(), unfortunately it also broke the --authfile/--insecure flags as they were removed.

This commit changes the approach to use ParseEarlyFlags() to parse flags that are already known before contacting the server, like --insecure & --authfile, so they can be used by the server while it executes GetGadgetInfo().

Fixes: d1bccf6c124f ("gadgets/run: Support eBPF parameters")
Fix https://github.com/inspektor-gadget/inspektor-gadget/issues/2225

## How to test 

0. Push a gadget to a insecure registry. https://github.com/inspektor-gadget/inspektor-gadget/issues/2225 contains instructions for it. In my case I pushed it to 192.168.1.150:5000/foo:latest

I manually tested the following:

### ig 

```bash
sudo -E ig run 192.168.1.150:5000/foo:latest --insecure
INFO[0000] Experimental features enabled                
RUNTIME.CONTAINERNAME                  PID                  PPID                 COMM             UID                  GID                 RETVAL
```

### ig + gadgectl using default unix socket

```bash
$ sudo -E ig daemon
INFO[0000] Experimental features enabled                
INFO[0000] starting Inspektor Gadget daemon at "unix:///var/run/ig/ig.socket"
...

$ sudo -E ./gadgetctl run 192.168.1.150:5000/foo:latest 
INFO[0000] Experimental features enabled                
RUNTIME.CONTAINERNAME                  PID                  PPID                 COMM             UID                  GID                 RETVAL
```

###  ig + gadgectl  using tcp 

```bash
$ sudo -E ig daemon -H tcp://127.0.0.1:1234
INFO[0000] Experimental features enabled                
INFO[0000] starting Inspektor Gadget daemon at "tcp://127.0.0.1:1234"

...
$ ./gadgetctl run 192.168.1.150:5000/foo:latest --remote-address tcp://127.0.0.1:1234 --insecure
INFO[0000] Experimental features enabled                
RUNTIME.CONTAINERNAME                  PID                  PPID                 COMM             UID                  GID                 RETVAL
```

### kubectl gadget 

```bash
$ ./kubectl-gadget run 192.168.1.150:5000/foo:latest --insecure -A 
INFO[0000] Experimental features enabled                
K8S.NODE              K8S.NAMESPACE         K8S.POD               K8S.CONTAINER        PID         PPID        COMM        UID         GID         RETVAL
```

### TODO
- [X] Add integration tests for for --insecure

### CODE REVIEW

**Comment by mauriciovasquezbernal** [comment]
I added an integration test for the --insecure flag, unfortunately doing the same for the --authfile isn't so easy as it'll require to configure [native basic auth](https://distribution.github.io/distribution/about/deploying/#native-basic-auth) and TLS certificates in the registry. I'll open an issue to handle this later on.

**Comment by mqasimsarfraz** [comment]
It seems that the `copier` pods keeps on restarting, perhaps we need pod restart policy `Never`. Maybe the failure is related to that? 🤔

**Comment by mauriciovasquezbernal** [comment]
> It seems that the `copier` pods keeps on restarting, perhaps we need pod restart policy `Never`. Maybe the failure is related to that? 🤔

I made this change, kicked the CI few times and it seems to be working fine now. Thanks!

File: integration/inspektor-gadget/run_insecure_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		{
+			// Wait for the image to be available in the registry
+			// TODO: Ideally it shouldn't be a delay, but there is no an easy way to
+			// wait for a completed command using kubectl run
+			Cmd: "sleep 2",
+		},
+	}
+	RunTestSteps(orasCpCmds, t)
+
+	// TODO: Ideally it should not depend on a real gadget, but we don't have a "test gadget" available yet.
```
Comment (burak-ok) [inline integration/inspektor-gadget/run_insecure_test.go:62]: Created issue https://github.com/inspektor-gadget/inspektor-gadget/issues/2255 to track this.

File: integration/inspektor-gadget/run_insecure_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		{
+			Cmd: fmt.Sprintf("kubectl run -n %s copier --image ghcr.io/oras-project/oras:v1.1.0 --restart=Never -- copy --to-plain-http %s/trace_open:%s %s:5000/trace_open:%s",
+				ns, *gadgetRepository, *gadgetTag, registryIP, *gadgetTag),
+		},
+		WaitUntilPodReadyCommand(ns, "copier"),
+		{
+			// Wait for the image to be available in the registry
+			// TODO: Ideally it shouldn't be a delay, but there is no an easy way to
+			// wait for a completed command using kubectl run
+			Cmd: "sleep 2",
```
Comment (burak-ok) [inline integration/inspektor-gadget/run_insecure_test.go:57]: Couldn't we use something like `kubectl wait --for=Succeeded` ? (Ref for `Succeeded` https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase )

Or we add `--rm` to the `kubectl run` command and wait for the pod deletion`

**Review by burak-ok — APPROVED** [review]
LGTM
In other places we also have `sleep` commands but we could maybe change that

File: integration/inspektor-gadget/run_insecure_test.go (hunk 2 above)
Comment (mauriciovasquezbernal) [reply integration/inspektor-gadget/run_insecure_test.go]: > Couldn't we use something like kubectl wait --for=Succeeded ?

Unfortunately it doesn't work. We can only wait on any of these conditions -> https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions

I decided to change it and use a job, that's a better fit for this use case. Can you check it again please?

**Review by mqasimsarfraz — APPROVED** [review]
I tested it and it works fine, thanks!

File: integration/inspektor-gadget/run_insecure_test.go (hunk 2 above)
Comment (burak-ok) [reply integration/inspektor-gadget/run_insecure_test.go]: See https://github.com/inspektor-gadget/inspektor-gadget/pull/2261

**Pull request merged**